server/odc-service/src/main/java/com/oceanbase/odc/service/task/service/StdTaskFrameworkService.java

```diff
@@ -0,0 +1,371 @@
+    }
+
+    private Specification<JobEntity> getRecentDaySpec(int days) {
+        return SpecificationUtil.columnLate("createTime", JobDateUtils.getCurrentDateSubtractDays(days));
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public JobDefinition getJobDefinition(Long id) {
+        JobEntity entity = find(id);
+        Class<? extends Task<?>> cl;
+        try {
+            cl = (Class<? extends Task<?>>) Class.forName(entity.getJobClass());
+        } catch (ClassNotFoundException e) {
+            throw new TaskRuntimeException(e);
+        }
+        return DefaultJobDefinition.builder()
+                .jobParameters(
+                        JsonUtils.fromJson(entity.getJobParametersJson(), new TypeReference<Map<String, String>>() {}))
+                .jobClass(cl).build();
+    }
+
+    @Override
+    @Transactional(rollbackFor = Exception.class)
+    public JobEntity save(JobDefinition jd) {
+        JobEntity jse = new JobEntity();
+        jse.setJobParametersJson(JsonUtils.toJson(jd.getJobParameters()));
+        jse.setJobPropertiesJson(JsonUtils.toJson(jd.getJobProperties()));
+        jse.setExecutionTimes(0);
+        jse.setJobClass(jd.getJobClass().getCanonicalName());
+        jse.setJobType(jd.getJobType());
+        jse.setStatus(JobStatus.PREPARING);
+        jse.setRunMode(taskFrameworkProperties.getRunMode().name());
+        return jobRepository.save(jse);
+    }
+
+    @Override
+    public void startSuccess(Long id, String executorIdentifier) {
+        JobEntity jobEntity = find(id);
+        Date currentDate = JobDateUtils.getCurrentDate();
+        jobEntity.setStatus(JobStatus.RUNNING);
+        jobEntity.setExecutorIdentifier(executorIdentifier);
+        // increment executionTimes
+        jobEntity.setExecutionTimes(jobEntity.getExecutionTimes() + 1);
+        // set current date as first heart time
+        jobEntity.setLastHeartTime(currentDate);
+        jobEntity.setStartedTime(currentDate);
+        if (jobEntity.getExecutorDestroyedTime() != null) {
+            jobEntity.setExecutorDestroyedTime(null);
+        }
+        jobRepository.updateJobExecutorIdentifierAndStatusById(jobEntity);
+    }
+
+
+    @Override
+    @Transactional(rollbackFor = Exception.class)
+    public void handleResult(TaskResult taskResult) {
+        if (taskResult.getJobIdentity() == null || taskResult.getJobIdentity().getId() == null) {
+            log.warn("Job identity is null");
+            return;
+        }
+        JobEntity je = find(taskResult.getJobIdentity().getId());
+        if (je.getStatus().isTerminated()) {
+            log.warn("Job {} is finished, ignore result", je.getId());
+            return;
+        }
+
+        updateJobScheduleEntity(taskResult);
+        if (resultHandleServices != null) {
+            resultHandleServices.forEach(r -> r.handle(taskResult));
+        }
+        if (publisher != null && taskResult.getStatus() != null && taskResult.getStatus().isTerminated()) {
```

Comment:
task change status at first, upload log at second.
so how to handle this? I destroy a pod when a pod is uploading log?

Reply:
in `BaseTask#doFinal`  method, make sure report terminate status before upload log completed